### PR TITLE
Use pkg-config to link against pam

### PR DESCRIPTION
**Pull request by euank**
Some distros, such as nixos, require more than just '-lpam' to locate
the library. Adding a naive pkg-config invocation to the pam FFI
crate allows pam_kanidm to build on my system, where before this change it did not.

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes
- [ ] ~book chapter included (if relevant)~ not relevant
- [ ] ~design document included (if relevant)~ not relevant

### CODE REVIEW

File: kanidm_unix_int/pam_kanidm/build.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+fn main() {
+    pkg_config::Config::new()
+        .atleast_version("1.3.0")
```
Comment (euank) [inline kanidm_unix_int/pam_kanidm/build.rs:3]: The minimum pam version was chosen somewhat arbitrarily.

I checked packaged versions from https://repology.org/project/pam/versions and just picked the number openSUSE has under the assumption that's the version that the code's been tested with.

Comment (yaleman) [reply kanidm_unix_int/pam_kanidm/build.rs:3]: Checks out from deb-ish land:

- Debian 10: 1.3.1
- Ubuntu 21.10: 1.3.1

**Comment by Firstyear** [comment]
@euank It looks like the CI failed, but maybe that's a fault of our CI env?

**Comment by euank** [comment]
Digging a little more, it looks like libpam only recently started including pkg-config files (https://github.com/linux-pam/linux-pam/pull/369).
The ubuntu `libpam-dev` package we install doesn't include one.

Maybe the right answer is to try pkg-config, but default to just spitting out `-lpam` and hoping?
I'll update this PR to that, and I expect that'll make CI happy at least.

**Comment by Firstyear** [comment]
Yeah I think that sounds like the best option :)

**Comment by euank** [comment]
What I've got now works with my NixOS setup, and with CI, so I think this is a reasonable approach to it

**Comment by Firstyear** [comment]
@euank Looks good! Can you also add yourself to https://github.com/kanidm/kanidm/blob/master/CONTRIBUTORS.md then I'll merge :)

**Comment by Firstyear** [comment]
Thank you very much @euank ! Really appreciate your help and work :)

**Comment by euank** [comment]
Thanks for accommodating my possibly slightly wonky distro setup I use, and of course for the project as a whole :)

**Comment by Firstyear** [comment]
No no, the more places that we test and run on, the better we become! So please continue to raise issues, give feedback and help out.

**Pull request merged**